### PR TITLE
Various fixes for Unix SslStream

### PR DESCRIPTION
**Pull request by bartonjs**
@vijaykota had put out a PR into the dev-sslstream branch which gets an SslStream acting as a client working enough to connect to https://www.microsoft.com/, send it a malformed request, and read the reply.

This PR is just a integrated version of #3365's non-experimental commits.

### CODE REVIEW

**Comment by bartonjs** [comment]
cc @CIPop, @DavidSh, @JosGuil @SidharthNabar, @PGavlin, @StephenToub, @rajansingh10 

I could technically count myself as the LGTM, but just because I used --author doesn't mean I necessarily did it honestly :).

File: src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs [hunk 1]
```diff
@@ -259,10 +260,10 @@ private SecurityStatus EncryptDecryptHelper(SafeDeleteContext securityContext, b
             {
                 securityContext.DangerousAddRef(ref gotReference);
                 inputHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
-                IntPtr inputPtr = Marshal.UnsafeAddrOfPinnedArrayElement(buffer, offset);
+                IntPtr inputPtr = Marshal.UnsafeAddrOfPinnedArrayElement(buffer, 0);
                 if (encrypt)
                 {
-                    resultSize = Interop.OpenSsl.Encrypt(securityContext.DangerousGetHandle(), inputPtr, size, buffer.Length);
+                    resultSize = Interop.OpenSsl.Encrypt(securityContext.DangerousGetHandle(), inputPtr, offset, size, buffer.Length);
```
Comment (stephentoub) [inline src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs]: While you're touching this function, consider adding a Debug.Fail to catch to print out the exception that happens.

File: src/System.Net.Security/src/System/Net/Windows/CertModule.cs [hunk 2]
```diff
@@ -154,6 +158,69 @@ internal override string[] GetRequestCertificateAuthorities(SafeDeleteContext se
             return issuers;
         }
 
+        //
```
Comment (stephentoub) [inline src/System.Net.Security/src/System/Net/Windows/CertModule.cs]: This was just moved to a different file and is identical to the original, right?

**Comment by stephentoub** [comment]
LGTM

File: src/System.Net.Security/src/System/Net/Windows/CertModule.cs (hunk 2 above)
Comment (bartonjs) [inline src/System.Net.Security/src/System/Net/Windows/CertModule.cs]: It seems to be that way; flipping back and forth between the add here and the delete in the other file, the only difference is here is override, there was static.

File: src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs (hunk 1 above)
Comment (bartonjs) [inline src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs]: > While you're touching this function, consider adding a Debug.Fail to catch to print out the exception that happens.

I'll leave that for @vijaykota and/or @rajansingh10.  I'd hope that the InternalError that it's returning would suggest that it's very unlikely; but wouldn't want to be responsible for the Debug.Fail that prints every time a remote host disconnects :).

**Comment by bartonjs** [comment]
Somehow I doubt this caused a catastrophic failure in tests:
```
02:15:41 Unhandled Exception: 
02:15:41    Cannot print exception string because Exception.ToString() failed.
02:15:42 
02:15:42 Unhandled Exception: 
02:15:42 Unhandled Exception: 
02:15:42    Cannot print exception string because Exception.ToString() failed.
02:15:42 
02:15:42    Cannot print exception string because Exception.ToString() failed.
```

**Comment by bartonjs** [comment]
@dotnet-bot test this please (do better this time, please)

File: src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs (hunk 1 above)
Comment (vijaykota) [inline src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs]: Will take care of this in a separate PR

File: src/System.Net.Security/src/System/Net/Windows/CertModule.cs (hunk 2 above)
Comment (vijaykota) [inline src/System.Net.Security/src/System/Net/Windows/CertModule.cs]: Yes.. only change is from static to override

**Comment by ellismg** [comment]
@bartonjs There are some problems with test runs now as I land globalization stuff.  Please stand by.

File: src/System.Net.Security/src/System/Net/_SecureChannel.cs [hunk 3]
```diff
@@ -249,7 +245,8 @@ private X509Certificate2 EnsurePrivateKey(X509Certificate certificate)
 
                 if (certEx != null)
                 {
-                    if (certEx.HasPrivateKey)
+                    // TODO: Fix after HasPrivateKey is fixed
+                    if (!String.IsNullOrEmpty(certEx.Subject)) //certEx.HasPrivateKey)
```
Comment (vijaykota) [inline src/System.Net.Security/src/System/Net/_SecureChannel.cs]: Undo changes in this line (this was part of the Hacks commit :))

Comment (bartonjs) [inline src/System.Net.Security/src/System/Net/_SecureChannel.cs]: Good thing it's had its share of CI issues :)

**Comment by stephentoub** [comment]
@dotnet-bot test this please (restarting after ICU merge)

**Comment by bartonjs** [comment]
@dotnet-bot test this please (some parts of the last run failed to download packages)

**Comment by bartonjs** [comment]
Looks like the globalization tests are failing on Linux; probably because it needs changes from master... working on it.

**Comment by ellismg** [comment]
@dotnet-bot test this please (restarting after @bartonjs has merged master into dev-sslstream to pick up globalization fixes).

File: src/System.Net.Security/src/System/Net/CertInterface.cs [hunk 4]
```diff
@@ -15,5 +15,7 @@ internal abstract class CertInterface
 
         internal abstract string[] GetRequestCertificateAuthorities(SafeDeleteContext securityContext);
 
+        internal abstract X509Store EnsureStoreOpened(bool isMachineStore);
+
     }
 }
```
Comment (CIPop) [inline src/System.Net.Security/src/System/Net/CertInterface.cs]: Nit: please add new line at EOF.

**Pull request merged**